File: linearsearch.c
/* Linear Search */

#include<stdio.h>
int linearsearch(int *a, int len, int data){
	int flag = -1;
	int i;
	for(i=0; i<len; i++){
		if(a[i] == data){
			flag = i;
			break;
		}
	}
	return flag;
}
int main(int argc, char** argv){
	int arr[] = {3,5,2,8,15,18,29,6,4};
	int length = sizeof(arr)/sizeof(arr[0]);
	int data;
	printf("Enter data to be searched\n");
	scanf("%d", &data);
	int result = linearsearch(arr, length, data);
	if(result == -1) printf("Element not found\n");
	else printf("Element found at %d\n", result);

	return 0;
}

File: linearsearch.py
def linearSearch(li, length, data):
	flag = -1
	for i in range(length):
		if li[i] == data:
			flag = i
			break
	return flag

def main():
	li = [3, 5, 2, 8, 15, 18,29, 6, 4]
	length = len(li)

	data = int(input("Enter data to be searched: "))

	result = linearSearch(li, length, data)
	if result == -1: 
		print("Element not found")
	else:
		print(f"Element found at {result}")

if __name__ == "__main__":
	main() 
